src/llm_tracekit/bedrock/invoke_agent.py

```diff
@@ -87,78 +192,166 @@ def record_invoke_agent_result_attributes(
+            response_model=result.foundation_model,
         )
 
 
 class InvokeAgentStreamWrapper(ObjectProxy):
-    """Wrapper for botocore.eventstream.EventStream"""
+    """
+    A wrapper for botocore.eventstream.EventStream that intercepts and processes
+    events from the Bedrock `invoke_agent` API to gather telemetry data.
+    It is designed to work robustly whether `enableTrace` is True or False.
+    """
 
     def __init__(
         self,
         stream: EventStream,
-        stream_done_callback: Callable[
-            [Optional[str], Optional[int], Optional[int]], None
-        ],
+        stream_done_callback: Callable[[AgentStreamResult], None],
         stream_error_callback: Callable[[Exception], None],
     ):
         super().__init__(stream)
-
         self._stream_done_callback = stream_done_callback
         self._stream_error_callback = stream_error_callback
-        self._content = None
+
+        self._content: Optional[str] = None
         self._usage_input_tokens: Optional[int] = None
         self._usage_output_tokens: Optional[int] = None
+        self._foundation_model: Optional[str] = None
+        self._inference_config_max_tokens: Optional[int] = None
+        self._inference_config_temperature: Optional[float] = None
+        self._inference_config_top_k: Optional[int] = None
+        self._inference_config_top_p: Optional[float] = None
+        self._chat_history: Optional[Tuple[List[Message], List[Choice]]] = None
+        self._tool_calls_buffer: List[ToolCall] = []
 
     def __iter__(self):
         try:
             for event in self.__wrapped__:
                 self._process_event(event)
                 yield event
 
-            self._stream_done_callback(
-                self._content, self._usage_input_tokens, self._usage_output_tokens
+            result = AgentStreamResult(
+                content=self._content,
+                usage_input_tokens=self._usage_input_tokens,
+                usage_output_tokens=self._usage_output_tokens,
+                foundation_model=self._foundation_model,
+                inference_config_max_tokens=self._inference_config_max_tokens,
+                inference_config_temperature=self._inference_config_temperature,
+                inference_config_top_k=self._inference_config_top_k,
+                inference_config_top_p=self._inference_config_top_p,
+                chat_history=self._chat_history,
+                tool_calls_buffer=self._tool_calls_buffer,
             )
+            self._stream_done_callback(result)
         except EventStreamError as exc:
             self._stream_error_callback(exc)
             raise
 
     def _process_usage_data(self, usage: Dict[str, int]):
-        if "inputTokens" in usage:
-            if self._usage_input_tokens is None:
-                self._usage_input_tokens = 0
+        if self._usage_input_tokens is None:
+            self._usage_input_tokens = 0
```

Comment:
I'm not sure the logic change is correct here, there is a difference between `usage_input_tokens = 0` and `usage_input_tokens = None` - if we don't have usage data we should keep it as `None`